### PR TITLE
Document the use of digest algorithm in fingerprint/2

### PR DESCRIPTION
**Pull request by dvic**

### CODE REVIEW

**Comment by dvic** [comment]
@josevalim not sure if it's needed, but we could switch to `sha1` for a bit more bits using

```elixir
    data = :erlang.term_to_binary({block, static})
    <<fingerprint::8*20>> = :crypto.hash(:sha, data)
    fingerprint
```

Not sure if LiveView is vulnerable in this way but it seems pretty [cheap](https://en.wikipedia.org/wiki/MD5#Collision_vulnerabilities) to find a MD5 collision these days.

**Comment by josevalim** [comment]
We don't use for crypto purposes, that's why md5 is fine. The values we use are not received or generated by the client. :)

File: lib/phoenix_live_view/engine.ex [hunk 1]
```diff
@@ -1006,6 +1006,9 @@ defmodule Phoenix.LiveView.Engine do
   ## Callbacks
 
   defp fingerprint(block, static) do
+    # We don’t check for collisions and therefore it is important the algorithm we
+    # use here guarantees a very low number of collisions.
```
Comment (josevalim) [inline lib/phoenix_live_view/engine.ex:1010]:
```suggestion
    # The fingerprint must be unique and we don’t check for collisions in the
    # Diff module as doing so would be expensive. Therefore it is important
    # that the algorithm we use here has a low number of collisions.
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**